### PR TITLE
Added Branch Guide (HTK) - resolves #3088 

### LINKED ISSUE #3088: The GitHub site needs a Mixed Reality Toolkit branch guide
**Does this affect the legacy HoloToolkit (master) or the Mixed Reality Toolkit (mrtk_release)?**
HoloToolkit | Mixed Reality Toolkit - Both

**Is your feature request related to a problem? Please describe.**
A fair few developers are confused by the multiple branches, what they mean and which branch is for which project.

**Describe the solution you'd like**
To resolve this, we need a simply "Branch Guide" to inform new developers approaching the toolkit what to expect.

Both the htk and MRTK branchs need to link to this guide

**Describe alternatives you've considered**
Telling people, but this is a repeating process

**Additional context**
Docs are a must!!

### PR DESCRIPTION
**Pull request by SimonDarksideJ**
Overview
Added a new Branch guide, needs replicating straight to the master and updated the homepage with a link

Changes
Fixes: #3088

### CODE REVIEW

File: BranchGuide.md [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+---
+## Feature Branches
+
+These relate to new components / features that are actively being developed for consideration to be included in the main development branch.
+These should not be considered stable and main contain breaking features, **use at your own risk**.
+
+> This list may not be up to date as development during the vNext beta is quite rapid, so simply use this as a guide and [ask in Slack](https://holodevelopersslack.azurewebsites.net/) regarding future features / capabilities.
+
+* [feature/LuminPlatform](https://github.com/Microsoft/MixedRealityToolkit-Unity/tree/feature/LuminPlatform) - (@StephenHodgson) Experimental Magic Leap device integration.
+* [feature/spatialAwarenessPlanes](https://github.com/Microsoft/MixedRealityToolkit-Unity/tree/feature/spatialAwarenessPlanes) - (@Keveleigh) - Extensions to the Spatial Awareness framework for  plane detection.
```
Comment (StephenHodgson) [inline BranchGuide.md]: If this feature is complete, then shouldn't continue to have a branch. Last I checked it made it back into development.

Comment (SimonDarksideJ) [reply BranchGuide.md]: rofl, the old feature was "complete" this is a "new" feature to complement the "other" feature.
But I can't comment on those "features" only document them.

File: BranchGuide.md [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+
+For example, here are some of the main area that are currently being worked on:
+
+* [feature/LuminPlatform](https://github.com/Microsoft/MixedRealityToolkit-Unity/tree/feature/LuminPlatform) - (@StephenHodgson) Experimental Magic Leap device integration.
+* [feature/spatialAwarenessPlanes](https://github.com/Microsoft/MixedRealityToolkit-Unity/tree/feature/spatialAwarenessPlanes) - (@davidkline-ms) - Extensions to the Spatial Awareness framework for  plane detection.
+* [feature/mrtk_interactable](https://github.com/Microsoft/MixedRealityToolkit-Unity/tree/feature/mrtk_interactable) - (@killerantz) New SDK components / features and ported features from the HTK
+* [feature/mrtk_packageManager](https://github.com/Microsoft/MixedRealityToolkit-Unity/tree/feature/mrtk_packageManager) - (@ryzngard) New MRTK delivery development (NuGet)
+* [feature_mrtk_state_sharing](https://github.com/Microsoft/MixedRealityToolkit-Unity/tree/feature_mrtk_state_sharing) - (@Railboy) Advanced networking / Sharing solution
+* [feature/mrtk_sharing](https://github.com/Microsoft/MixedRealityToolkit-Unity/tree/feature/mrtk_sharing) - (@Sean0ng) Photon Based networking / sharing solution
```
Comment (StephenHodgson) [inline BranchGuide.md]:
```suggestion
* [feature/sharing_photon_preview](https://github.com/Microsoft/MixedRealityToolkit-Unity/tree/feature/sharing_photon_preview) - (@Sean0ng) Photon Based networking / sharing solution
```

Per our MRTK Sharing Meeting

**Comment by SimonDarksideJ** [comment]
Awaiting  #3089 to finalise updates

**Comment by SimonDarksideJ** [comment]
Replaced by #3135

**Pull request closed without merging**